src/hooks/useSetColumnDefaultSortOrder.ts

```diff
@@ -0,0 +1,15 @@
+import { SortDirection } from "gql/generated/types";
+import { ColumnProps } from "antd/es/table";
+
+export const useSetColumnDefaultSortOrder = <T>(
+  columns: ColumnProps<T>[],
+  category: string,
+  direction: string
+) => {
+  const targetColumnHeader = columns.find(({ key }) => key === category);
+  if (targetColumnHeader) {
+    // eslint-disable-next-line no-param-reassign
+    targetColumnHeader.defaultSortOrder =
+      direction === SortDirection.Asc ? "ascend" : "descend";
+  }
```

Comment:
Im reluctant on modifying the param passed in  directly. The only reason this works here is because objects are passed in by reference. When columns is [initially declared](https://github.com/evergreen-ci/spruce/pull/182/files#diff-345b351a35b76219866aa63f353fa764R213) its declared as a `const` which implies it should be immutable. By doing this you are breaking that assumption.  I would instead have this hook return a duplicate object and reassign `columns` with the returned value. The `no-param-reassign` rule is there to help avoid that since mutating objects in this manner could lead to some unintended side effects.

Reply:
very good point